packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -0,0 +1,374 @@
+    public multiple: boolean;
+
+    @attr({ attribute: "open", mode: "boolean" })
+
+    @observable
+    public open: boolean;
+    private openChanged() {
+        this.updateButtonPartAttr();
+    }
+
+    @observable
+    public defaultSlottedNodes: Node[];
+
+    @observable
+    public button: HTMLElement;
+
+    @observable
+    public selectedValue: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public listbox: HTMLElement[];
+    private listboxChanged() {
+        if (this.$fastController.isConnected) {
+            this.applyListboxControllerCode();
+        }
+    }
+
+    /**
+     * @internal
+     */
+    @observable
+    public options: Option[];
+    private optionsChanged() {
+        if (this.$fastController.isConnected) {
+            //
+        }
+    }
+
+    // TODO: This needs to change to support multiple values
+    public value: string = "Selected Value"; // Map to proxy element.
+    public valueChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.value = this.value;
+        }
+    }
+
+    private activeOptionIndex: number = 0;
+
+    /**
+     * Set to true when the component has constructed
+     */
+    private constructed: boolean = false;
+
+    constructor() {
+        super();
+        this.constructed = true;
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        this.registerButtonSlotChange();
+
+        // We won't get a slotchange for event for parts that were not replaced
+        // by user-provided parts, so apply their controller code here.
+        this.applyButtonControllerCode();
+        this.applyListboxControllerCode();
+
+        this.updateForm();
+    }
+
+    private updateForm(): void {
+    }
+
+    
+    public keypressHandlerButton = (e: KeyboardEvent): void => {
+        super.keypressHandler(e);
+        switch (e.keyCode) {
+            case KeyCodes.space:
+                this.open = !this.open;
+                console.log("Call one")
+                setTimeout(() => this.setFocusOnOption(), 0);
+                break;
+            case KeyCodes.enter:
+                this.open = !this.open;
+                console.log("Call two")
+                setTimeout(() => this.setFocusOnOption(), 0);
+                e.preventDefault(); // Enter also causes 'click' to fire for <button>s.  Prevent that so we don't immediately revert the change to this.open.
+                break;
+            case KeyCodes.tab:
+                this.setFocusOnOption();
+                break;
+        }
+    }
+
+    /**
+     * Handle keyboard interactions for listbox
+     */
+    public keypressHandlerListbox = (e: KeyboardEvent): void => {
+        super.keypressHandler(e);
+        
+        // Don't scroll the page for arrow keys, and spacebar.
+        e.preventDefault();
+        switch (e.keyCode) {
+            case KeyCodes.arrowDown:
```

Comment:
Handling navigation across the options should be handled by the "listbox" component IMO.